extensions/microsoft-authentication/src/extension.ts

```diff
@@ -36,7 +36,12 @@ async function initAzureCloudAuthProvider(context: vscode.ExtensionContext, tele
 		settingValue += '/';
 	}
 
-	const azureEnterpriseAuthProvider = new AzureActiveDirectoryService(context, uriHandler, tokenStorage, settingValue);
+	const azureEnterpriseAuthProvider = new AzureActiveDirectoryService(
+		vscode.window.createOutputChannel(vscode.l10n.t('Microsoft Sovereign Cloud Authentication'), { log: true }),
```

Comment:
The sovereign auth provider gets recreated when the endpoint setting changes--would having a new output channel of the same name created make VSCode upset, or would it just keep flowing like normal?

Reply:
At least from testing, it keeps things flowing like normal. I'll let @sandy081 comment on the correct way to use this API

Reply:
Internally it reuses the same logger because they are mapped to name. I would recommend to dispose the previously created log channels so that they won't be dangling around.